parser-typechecker/unison/Main.hs

```diff
@@ -253,88 +162,91 @@ upgradeCodebase mcodepath =
-prepareTranscriptDir cbFormat inFork mcodepath = do
+prepareTranscriptDir :: CodebaseFormat -> ShouldForkCodebase -> Maybe FilePath -> IO FilePath
+prepareTranscriptDir cbFormat shouldFork mcodepath = do
   tmp <- Temp.getCanonicalTemporaryDirectory >>= (`Temp.createTempDirectory` "transcript")
   let cbInit = cbInitFor cbFormat
-  if inFork then
-    getCodebaseOrExit cbFormat mcodepath >> do
-    path <- Codebase.getCodebaseDir mcodepath
-    PT.putPrettyLn $ P.lines [
-      P.wrap "Transcript will be run on a copy of the codebase at: ", "",
-      P.indentN 2 (P.string path)
-      ]
-    Path.copyDir (Codebase.codebasePath cbInit path) (Codebase.codebasePath cbInit tmp)
-  else do
-    PT.putPrettyLn . P.wrap $ "Transcript will be run on a new, empty codebase."
-    void $ Codebase.openNewUcmCodebaseOrExit cbInit "main.transcript" tmp
+  case shouldFork of
+    UseFork -> do
+      getCodebaseOrExit cbFormat mcodepath
+      path <- Codebase.getCodebaseDir mcodepath
+      PT.putPrettyLn $ P.lines [
+        P.wrap "Transcript will be run on a copy of the codebase at: ", "",
+        P.indentN 2 (P.string path)
+        ]
+      Path.copyDir (Codebase.codebasePath cbInit path) (Codebase.codebasePath cbInit tmp)
+    DontFork -> do
+      PT.putPrettyLn . P.wrap $ "Transcript will be run on a new, empty codebase."
+      void $ Codebase.openNewUcmCodebaseOrExit cbInit "main.transcript" tmp
   pure tmp
 
 runTranscripts'
   :: CodebaseFormat
   -> Maybe FilePath
   -> FilePath
-  -> [String]
+  -> NonEmpty String
   -> IO Bool
 runTranscripts' codebaseFormat mcodepath transcriptDir args = do
   currentDir <- getCurrentDirectory
-  case args of
-    args@(_:_) -> do
-      for_ args $ \arg -> case arg of
-        md | isMarkdown md -> do
-          parsed <- TR.parseFile arg
-          case parsed of
-            Left err ->
-              PT.putPrettyLn $ P.callout "❓" (
-                P.lines [
-                  P.indentN 2 "A parsing error occurred while reading a file:", "",
-                  P.indentN 2 $ P.string err])
-            Right stanzas -> do
-              configFilePath <- getConfigFilePath mcodepath
-              (closeCodebase, theCodebase) <- getCodebaseOrExit codebaseFormat $ Just transcriptDir
-              mdOut <- TR.run transcriptDir configFilePath stanzas theCodebase
-              closeCodebase
-              let out = currentDir FP.</>
-                         FP.addExtension (FP.dropExtension arg ++ ".output")
-                                         (FP.takeExtension md)
-              writeUtf8 out mdOut
-              putStrLn $ "💾  Wrote " <> out
-        wat ->
-              PT.putPrettyLn $ P.callout "❓" (
-                P.lines [
-                  P.indentN 2 "Unrecognized command, skipping:", "",
-                  P.indentN 2 $ P.string wat])
-      pure True
-    [] ->
-      pure False
+  let (markdownFiles, invalidArgs) = NonEmpty.partition isMarkdown args
```

Comment:
Small refactor of this function to support NonEmpty list & to batch all invalid files for error messaging.